apps/arweave/src/ar_chunk_storage.erl

```diff
@@ -809,15 +810,26 @@ generate_entropy_keys(RewardAddr, Offset, SubChunkStart) ->
 	[ar_replica_2_9:get_entropy_key(RewardAddr, Offset, SubChunkStart)
 			| generate_entropy_keys(RewardAddr, Offset, SubChunkStart + SubChunkSize)].
 
+store_entropy(
+		_Entropies, PaddedEndOffset, _SubChunkStartOffset,
+		RangeEnd, _Keys, _RewardAddr, N, WaitN)  when PaddedEndOffset > RangeEnd ->
+	%% The amount of entropy generated per partition is slightly more than the amount needed.
+	%% So at the end of a partition we will have finished processing chunks, but still have
+	%% some entropy left. In this case we stop the recursion early and wait for the writes
+	%% to complete.
+	wait_store_entropy_processes(WaitN),	
+	{ok, N};
 store_entropy(
 		Entropies, PaddedEndOffset, SubChunkStartOffset,
-		Partition, Keys, RewardAddr, N, WaitN) ->
+		RangeEnd, Keys, RewardAddr, N, WaitN) ->
 	case take_and_combine_entropy_slices(Entropies) of
 		{<<>>, []} ->
+			%% We've finished processing all the entropies, wait for the writes to complete.
 			wait_store_entropy_processes(WaitN),
 			{ok, N};
 		{ChunkEntropy, Rest} ->
-			true = ar_replica_2_9:get_entropy_partition(PaddedEndOffset) == Partition,
```

Comment:
Could you explain this change? The way I see it the original assertion should hold because store_entropy only takes the entropies corresponding to a single partition in the input.

Reply:
Yep! The intent of the assertion is the same (unless Ive made a mistake)

Previously Partition was calculated from PartitionEnd and then passed into the function and, I believe, only used in the assertion. 

Now I pass in PartitionEnd directly so I can use it in a guard to stop the recursion. So the assertion *should* still be asserting the same logic (one partition of entropy) but it's doing it using PartitionEnd instead of Partition.

Reply:
Ah, right, thank you. What do you think about keeping the previous assertion too?

Reply:
Like re add the Partition argument?

Reply:
Ah, my bad, nevermind, I see RangeEnd replaced Partition so the assertion is changed accordingly.

Reply:
Thanks!